src/core/labeling/rules/qgslabelingengineruleregistry.cpp

```diff
@@ -0,0 +1,81 @@
+/***************************************************************************
+    qgslabelingengineruleregistry.cpp
+    ---------------------
+  Date                 : August 2024
+  Copyright            : (C) 2024 by Nyall Dawson
+  Email                : nyall dot dawson at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgslabelingengineruleregistry.h"
+#include "qgslabelingenginerule.h"
+#include "qgslabelingenginerule_impl.h"
+#include <geos_c.h>
+
+QgsLabelingEngineRuleRegistry::QgsLabelingEngineRuleRegistry()
+{
+#if GEOS_VERSION_MAJOR>3 || ( GEOS_VERSION_MAJOR == 3 && GEOS_VERSION_MINOR>=10 )
+  addRule( new QgsLabelingEngineRuleMinimumDistanceLabelToFeature() );
+  addRule( new QgsLabelingEngineRuleMaximumDistanceLabelToFeature() );
+  addRule( new QgsLabelingEngineRuleMinimumDistanceLabelToLabel() );
+#endif
```

Comment:
If we're conditionally adding rules to the registry based on GEOS version, that'll mean a project shared across multiple users with different GEOS versions (or projects processed on a server) will result in rule loss. Is that something you've thought about and are we OK with that? 

An alternative would be to add a QLabel note in the rule widget UI saying 'this rule will be ignored on your system' or something.

Reply:
Ok, I've gone with:

- grey out the rule in the rules list
- show an explanatory tooltip
- don't allow users to create new versions of that rule